lib/dns.js

```diff
@@ -137,12 +137,16 @@ function onlookupall(err, addresses) {
 // Easy DNS A/AAAA look up
 // lookup(hostname, [options,] callback)
 const validFamilies = [0, 4, 6];
+const dnsCache = { __proto__: null };
 function lookup(hostname, options, callback) {
   let hints = 0;
   let family = 0;
   let all = false;
   let verbatim = getDefaultVerbatim();
 
+  const cacheKey = `${hostname}_${family || 'default'}`;
```

Comment:
IMO this (and the code added before the if(!hostname) should move to before `const req= new GetAddrInfoReqWrap()`.. The hostname variable should be used only after it passes validation.

Reply:
The cache key should probably include if options.all is true or not

Reply:
I thought if there's a dns key for that domain that means it passed the validation earlier so there was no need for validating the hostname again. I could be wrong though

Also having the `all` in dns key makes a lot of sense I will add that.